.cirrus.yml

```diff
@@ -7,6 +7,15 @@ task:
     PATH: "$CIRRUS_WORKING_DIR/bin:$CIRRUS_WORKING_DIR/bin/cache/dart-sdk/bin:$PATH"
 
   git_fetch_script: git fetch origin
+  pub_cache:
+    folder: $HOME/.pub-cache
+    fingerprint_script: find . -name 'pubspec.yaml' -exec grep -H 'PUBSPEC CHECKSUM' {} +
```

Comment:
Use `$CIRRUS_WORKING_DIR` instead of `.`, just to be more explicit about where it's executing.

Also, this would be more efficient as a pipe to xargs:
```bash
find "$CIRRUS_WORKING_DIR" -name pubspec.yaml | xargs grep -H 'PUBSPEC CHECKSUM'
```
-exec invokes a new grep for each item found, but xargs will invoke grep on as many files as it can at once, which is more efficient.  In fact, the following is even more efficient:
```bash
grep -r --include=pubspec.yaml 'PUBSPEC CHECKSUM' "$CIRRUS_WORKING_DIR"
```
Since it only invokes one executable once.

Reply:
I read that the `+` on the end of find would pass all filenames once, but I'm not familiar with these so I don't know if it's exactly the same.

I'll change to what you suggest last though - I did look for how to do it all with grep and somehow failed to find that!

(I'll wait until I've landed the other change before fixing this, so it's in less places, and it's going to conflict anyway).